python/cudf/cudf/utils/utils.py

```diff
@@ -70,7 +70,14 @@ def scalar_broadcast_to(scalar, size, dtype=None):
         return column_empty(size, dtype=dtype, masked=True)
 
     if isinstance(scalar, pd.Categorical):
-        return scalar_broadcast_to(scalar.categories[0], size).astype(dtype)
+        if dtype is None:
```

Comment:
The logic for handling `dtype=None` appears right below this snippet. If we moved this to below line 86, does that break anything?

Reply:
If i understand you correctly you mean moving this section to the section that calls `to_cudf_compatible_scalar` . Unfortunately that fails with pandas with below error. 
```python
import pandas as pd
from cudf.utils.dtypes import to_cudf_compatible_scalar

val = pd.Categorical(values='a',categories=['a'])
# below line errors
# val = pd.api.types.pandas_dtype(type(val)).type(val)

to_cudf_compatible_scalar(val)
```

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-28-e53c389d5da0> in <module>
      6 #val = pd.api.types.pandas_dtype(type(val)).type(val)
      7 
----> 8 to_cudf_compatible_scalar(val)

/nvme/0/vjawa/dev/rapids/cudf/python/cudf/cudf/utils/dtypes.py in to_cudf_compatible_scalar(val, dtype)
    170         )
    171 
--> 172     val = pd.api.types.pandas_dtype(type(val)).type(val)
    173 
    174     if dtype is not None:

/nvme/0/vjawa/dev/rapids/compose/etc/conda/cuda_10.2/envs/notebooks/lib/python3.7/site-packages/pandas/core/dtypes/common.py in pandas_dtype(dtype)
   2065         return npdtype
   2066     elif npdtype.kind == "O":
-> 2067         raise TypeError("dtype '{}' not understood".format(dtype))
   2068 
   2069     return npdtype

TypeError: dtype '<class 'pandas.core.arrays.categorical.Categorical'>' not understood
```